cmake/GenerateBuiltins.cmake

```diff
@@ -72,13 +72,19 @@ function(target_ll_to_cpp llFileName bit os_name resultFileName)
     set_source_files_properties(${resultFileName} PROPERTIES GENERATED true)
 endfunction()
 
-function(dispatch_ll_to_cpp llFileName os_name resultFileName)
-    set(inputFilePath builtins/${llFileName}.ll)
-    set(output ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/builtins-${llFileName}.cpp)
+function(dispatch_to_cpp os_name resultFileName)
+    set(inputFilePath builtins/dispatch.c)
+    set(DISP_TYPE -DREGULAR)
+    set(DISP_NAME "dispatch")
+    if (${os_name} STREQUAL "macos")
+        set(DISP_TYPE -DMACOS)
+        set(DISP_NAME "dispatch-macos")
+    endif()
+    set(output ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/builtins-${DISP_NAME}.cpp)
     add_custom_command(
         OUTPUT ${output}
-        COMMAND ${M4_EXECUTABLE} -DLLVM_VERSION=${LLVM_VERSION} ${inputFilePath}
-            | \"${Python3_EXECUTABLE}\" bitcode2cpp.py ${inputFilePath} --type=dispatch --os=${os_name} --llvm_as ${LLVM_AS_EXECUTABLE} --opaque_flags="${LLVM_TOOLS_OPAQUE_FLAGS}"
+        COMMAND ${CLANGPP_EXECUTABLE} -x c ${ISPC_OPAQUE_FLAGS} ${DISP_TYPE} -march=x86-64 -mtune=generic -O2 -emit-llvm -S -c ${inputFilePath} -o -
```

Comment:
I have this error on ARM macOS without cross-build enabled:
```
[  5%] Generating builtins-dispatch-macos.cpp
clang++: error: unsupported argument 'x86-64' to option '-march='
[  7%] Generating builtins-dispatch.cpp
clang++: error: unsupported argument 'x86-64' to option '-march='
```
which is not surprising.

Also, looks like we are missing ARM builds in CI. We do have hosted ARM Linux runner, we should enabled a ARM-only and full ARM build there. We used to have Travis to test ARM, but it's not running anymore, I'm not sure what happened to it.